### ISSUE TITLE
Building with JDK 17 fails

### ISSUE THREAD

**Issue by arichiardi** [xtdb/crux#1723]
Hi there, wanted to report that I tried to build XTDB with the following setup and JDK 17:

`sdk default java  17.0.2-open`

and this is what I get

```
$ ./lein-sub install
--core
Compiling 30 source files to /home/cokap/git/xtdb/core/target/classes
Compiling xtdb.main
Created /home/cokap/git/xtdb/core/target/xtdb-core-dev-SNAPSHOT.jar
Created /home/cokap/git/xtdb/core/target/sources/xtdb-core-dev-SNAPSHOT-sources.jar
No tools.jar found in system or specified in project, cannot run javadoc.
```

It seems like some command in the chain is trying to access the [removed lib/tools.jar](https://docs.oracle.com/en/java/javase/17/migrate/migrating-jdk-8-later-jdk-releases.html#GUID-A78CC891-701D-4549-AA4E-B8DD90228B4B) and the build is failing.

**Comment by jarohen** [comment]
Hey @arichiardi - I suspect this is `lein javadoc`. We tend to run XT against Java 8 as our minimum supported version, if that helps - I think it's as early as JDK9 that tools.jar was removed. But yes, one to fix!

**Comment by refset** [comment]
This is how we run `javadoc` using JDK17 in CircleCI https://github.com/xtdb/xtdb/blob/11377633579218866e81e4953348801eb8eedab6/.circleci/config.yml#L201-L217

Cross-linking with the original report: https://juxt-oss.zulipchat.com/#narrow/stream/251593-xtdb-dev/topic/Installing.20jars/near/276091145

**Comment by arichiardi** [comment]
Yeah the `javadoc` cli works but if you try and lauch it via classpath, you won't find it anymore.

**Comment by arichiardi** [comment]
@jarohen I just dug in 

> Hey @arichiardi - I suspect this is `lein javadoc`. We tend to run XT against Java 8 as our minimum supported version, if that helps - I think it's as early as JDK9 that tools.jar was removed. But yes, one to fix!

The message I am seeing matches [this piece of code](https://github.com/davidsantiago/lein-javadoc/blob/3e0fd1b3ac85c1cbec2fd6f5f710284e258508ef/src/leiningen/javadoc.clj#L98) so I guess you are right.

**Comment by refset** [comment]
I just hit this myself (forgot to switch back to JDK8 :sweat_smile:) - perhaps we are safe to remove the `lein-javadoc` plugin altogether?

i.e.
https://github.com/xtdb/xtdb/blob/ba5aced2cc7ed5298ddf0b83b45db7617f732387/core/project.clj#L4

**Comment by jarohen** [comment]
@refset we still use it for the various modules through the deploy process, so can't just remove it, unfortunately. this then becomes a javadoc JAR that gets uploaded with the main JAR and source JAR to Maven Central - if we were to remove the plugin we'd need to code this up separately, which may not play as nicely with `lein deploy`. needs some thought!

J

**Comment by arichiardi** [comment]
I think the best of option is to patch `lein javadoc`. However, there is no alternative in classes for JDK 17 for that.

The alternative needs to be coded up like you folks did in CI.

I guess one other thing we can/should do is to mention that at the moment only JDK 8 can build XTDB in the development documentation.
